core/controller/sync.ts

```diff
@@ -6,51 +6,36 @@ import { BaseController } from 'core/controller/baseController';
 import PeerService from 'core/service/peer';
 import BlockService from 'core/service/block';
 import { logger } from 'shared/util/logger';
-import { messageON } from 'shared/util/bus';
 import BlockController from 'core/controller/block';
-
-const START_PEER_REQUEST = 10000;
-const START_SYNC_BLOCKS = 15000;
+import PeerRepository from 'core/repository/peer';
 
 export class SyncController extends BaseController {
 
-    constructor() {
-        super();
-        // TODO remove after test
-        setTimeout(() => {
-            messageON('EMIT_REQUEST_PEERS', {});
-        }, START_PEER_REQUEST);
-        setTimeout(() => {
-            messageON('EMIT_SYNC_BLOCKS', {});
-        }, START_SYNC_BLOCKS);
-    }
-
-    // TODO call after several minutes or by interval
     @ON('EMIT_REQUEST_PEERS')
-    async requestPeers(): Promise<void> {
+    requestPeers(): void {
         SyncService.requestPeers();
     }
 
     @ON('REQUEST_PEERS')
-    async sendPeers(action: { peer: Peer }): Promise<void> {
+    sendPeers(action: { peer: Peer }): void {
         const { peer } = action;
         SyncService.sendPeers(peer);
     }
 
     @ON('RESPONSE_PEERS')
-    async connectNewPeers(action: { data: { peers } }): Promise<void> {
+    connectNewPeers(action: { data: { peers } }): void {
         const { data } = action;
         SyncService.connectNewPeers(data.peers);
     }
 
     @ON('EMIT_REQUEST_COMMON_BLOCKS')
-    async emitRequestCommonBlocks(block: { id: string, height: number }) {
+    emitRequestCommonBlocks(block: { id: string, height: number }): void {
         logger.debug(`[Controller][Sync][emitRequestCommonBlocks]}`);
         SyncService.requestCommonBlocks(block);
     }
 
     @ON('REQUEST_COMMON_BLOCKS')
-    async checkCommonBlocks(action: { data: { block: { id: string, height: number } }, peer: Peer }) {
+    checkCommonBlocks(action: { data: { block: { id: string, height: number } }, peer: Peer }): void {
```

Comment:
how about to use `types` when we have a lot of arguments? 
for example you have `type Peer = { peer: Peer }` and for your purpose it will be: `type CheckCommonBlocksProps = Peer & {action: { data: { block: { id: string, height: number } }}`

Reply:
type Peer = { peer: Peer } bad example